suite-common/wallet-utils/src/__tests__/accountUtils.test.ts

```diff
@@ -337,4 +340,114 @@ describe('account utils', () => {
+        getWalletAccount({ symbol: 'matic', accountType: 'legacy' }),
+        getWalletAccount({ symbol: 'matic', accountType: 'normal' }),
+        getWalletAccount({ symbol: 'matic', accountType: 'ledger' }),
+        getWalletAccount({ symbol: 'sol', accountType: 'normal', empty: true, visible: false }),
+        getWalletAccount({ symbol: 'sol', accountType: 'ledger' }),
+        getWalletAccount({
+            symbol: 'sol',
+            accountType: 'ledger',
+            empty: true,
+            visible: false,
+        }),
+    ];
+
+    type RunFilterReceiveAccountsTestParams = {
+        isDebug?: boolean;
+        receiveNetwork?: string;
+        deviceState?: `${string}@${string}:${number}`;
+    };
+
+    const runFilterReceiveAccouns = ({
+        isDebug = true,
+        receiveNetwork = 'eth',
+        deviceState = '1stTestnetAddress@device_id:0',
+    }: RunFilterReceiveAccountsTestParams) => {
+        const device = getSuiteDevice({
+            unavailableCapabilities: {
+                dash: 'no-support',
+            },
+            state: deviceState,
+        });
+        const unavailableCapabilities = device?.unavailableCapabilities ?? {};
+
+        const receiveNetworks = networksCollection.filter(
+            (n: Network) =>
+                n.symbol === receiveNetwork &&
+                !unavailableCapabilities[n.symbol] &&
+                ((n.isDebugOnlyNetwork && isDebug) || !n.isDebugOnlyNetwork),
+        );
+
+        return filterReceiveAccounts({
+            accounts,
+            deviceState: device.state,
+            receiveNetwork,
+            isDebug,
+            receiveNetworks,
+        });
+    };
+
+    it('filterReceiveAccounts: isDebug', () => {
+        const filteredAccounts = [
+            getWalletAccount({ symbol: 'eth', accountType: 'legacy' }),
+            getWalletAccount({ symbol: 'eth', accountType: 'normal' }),
+            getWalletAccount({ symbol: 'eth', accountType: 'ledger' }),
+        ];
+
+        expect(runFilterReceiveAccouns({})).toEqual(filteredAccounts);
+    });
+
+    it('filterReceiveAccounts: isNotDebug', () => {
+        const filteredAccounts = [getWalletAccount({ symbol: 'eth', accountType: 'normal' })];
+
+        expect(runFilterReceiveAccouns({ isDebug: false })).toEqual(filteredAccounts);
+    });
+
+    it('filterReceiveAccounts: isNotSameDevice', () => {
+        expect(runFilterReceiveAccouns({ deviceState: '2ndTestnetAddress@device_id:0' })).toEqual(
+            [],
+        );
+    });
+
+    it('filterReceiveAccounts: isAnotherNetwork', () => {
+        const filteredAccounts = [
+            getWalletAccount({ symbol: 'btc', accountType: 'coinjoin' }),
+            getWalletAccount({ symbol: 'btc', accountType: 'taproot' }),
+            getWalletAccount({ symbol: 'btc', accountType: 'legacy' }),
+            getWalletAccount({ symbol: 'btc', accountType: 'segwit' }),
+            getWalletAccount({ symbol: 'btc', accountType: 'ledger' }),
```

Comment:
Could add a non btc account here, so you can test that it has been filtered out.
Analogically to case of `filterReceiveAccounts: isDebug`

Reply:
But I'm filtering them from a list of all the accounts I have declared above